Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -354,32 +403,55 @@ namespace AzToolsFramework
                 {
                     // TODO: check that the file actually exists
                     m_bookmarkfileName = bookmarkComponent->GetLocalBookmarksFileName();
+                    LoadViewBookmarks();
+
+                    for (int i = 0; i < m_localBookmarks.size(); i++)
+                    {
+                        AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
+                        registry->SetObject(finalPath, m_localBookmarks.at(i));
+                    }
+
+                    return false;
                 }
                 else // if the field is empty we don't have a file linked to the prefab, so we create one and we save it in the
                      // component
                 {
                     m_bookmarkfileName = GenerateBookmarkFileName();
+
+                    // Initialize default locations to 0
+                    for (int i = 0; i < s_numOfDefaultLocationsInLevel; i++)
+                    {
+                        AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+                        registry->SetObject(finalPath, ViewBookmark());
+                    }
+
+                    //SaveBookmarkSettingsFile();
```

Comment:
update: Should this commented line be removed?

Reply:
Already removed, just haven't pushed the changes :P